flink-runtime/src/test/java/org/apache/flink/runtime/util/JarFileCreatorTest.java

```diff
@@ -102,15 +104,62 @@ public void TestAnonymousInnerClassTrick4() throws Exception {
 		jfc.addClass(NestedAnonymousInnerClass.class)
 			.createJarFile();
 
-		Set<String> ans = new HashSet<String>();
+		Set<String> ans = new HashSet<>();
 		ans.add("org/apache/flink/runtime/util/jartestprogram/NestedAnonymousInnerClass.class");
 		ans.add("org/apache/flink/runtime/util/jartestprogram/NestedAnonymousInnerClass$1$1.class");
 		ans.add("org/apache/flink/runtime/util/jartestprogram/NestedAnonymousInnerClass$1.class");
 		ans.add("org/apache/flink/runtime/util/jartestprogram/NestedAnonymousInnerClass$A.class");
 
 		Assert.assertTrue("Jar file for Anonymous Inner Class is not correct", validate(ans, out));
 
-		out.delete();
+		Assert.assertTrue(out.delete());
+	}
+
+	@Ignore // this is currently not supported (see FLINK-9520)
+	@Test
+	public void testFilterWithMethodReference() throws Exception {
```

Comment:
Did these also not work when they were still in the java8 module?

Reply:
No this has never worked. I just noticed this bug when I looked at the test. What is this `JarFileCreator` good for actually?

Reply:
It was a util for programmatically creating Jar files. I think it's only used in tests right now, for example `ClassLoaderUtilsTest`.